test/libethereum/blockchainHelper.cpp

```diff
@@ -0,0 +1,239 @@
+	dev::eth::BlockChain *m_blockchain;
+	dev::eth::State m_state;	
+	std::vector<TestBlock> m_testBlocks;
+};
+
+std::string const c_genesisInfoTest =
+R"ETHEREUM(
+{
+	"a94f5374fce5edbc8e2a8697c15331677e6ebf0b": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+	"0000000000000000000000000000000000000001": { "wei": "1" },
+	"0000000000000000000000000000000000000002": { "wei": "1" },
+	"0000000000000000000000000000000000000003": { "wei": "1" },
+	"0000000000000000000000000000000000000004": { "wei": "1" },
+	"dbdbdb2cbd23b783741e8d7fcf51e459b497e4a6": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+	"e6716f9544a56c530d868e4bfbacb172315bdead": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+	"b9c015918bdaba24b4ff057a92a3873d6eb201be": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+	"1a26338f0d905e295fccb71fa9ea849ffa12aaf4": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+	"2ef47100e0787b915105fd5e3f4ff6752079d5cb": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+	"cd2a3d9f938e13cd947ec05abc7fe734df8dd826": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+	"6c386a4b26f73c802f34673f7248bb118f97424a": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+	"e4157b34ea9615cfbde6b4fda419828124b70c78": { "wei": "1606938044258990275541962092341162602522202993782792835301376" },
+}
+)ETHEREUM";
+
+BOOST_AUTO_TEST_SUITE(blockchainhelper)
+
+BOOST_AUTO_TEST_CASE(blockChainReload)
+{
+	std::string genesisTemp = dev::eth::c_genesisInfo;
+
+	//A trick for blockchain to load changed genesis state
+	std::string *genesis_ptr;
+	genesis_ptr = const_cast<std::string*>( &dev::eth::c_genesisInfo );
+	*genesis_ptr = c_genesisInfoTest;
+
+	TestBlockChain blChain;
+	json_spirit::mObject trJson;
+	trJson["data"] = "";
+	trJson["gasLimit"] = "0x59d8";
+	trJson["gasPrice"] = "0x01";
+	trJson["nonce"] = "0x00";
+	trJson["secretKey"] = "45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8";
+	trJson["to"] = "d2571607e241ecf590ed94b12d87c94babe36db6";
+	trJson["value"] = "0x0a";
+
+	TestBlock block;
+	TestTransaction tr(trJson);
+	block.addTransaction(tr);
+	blChain.addBlock(block);
+
+	json_spirit::mObject trJson2;
+	trJson2["data"] = "";
+	trJson2["gasLimit"] = "0x59d8";
+	trJson2["gasPrice"] = "0x01";
+	trJson2["nonce"] = "0x01";
+	trJson2["secretKey"] = "45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8";
+	trJson2["to"] = "d2571607e241ecf590ed94b12d87c94babe36db6";
+	trJson2["value"] = "0x0a";
+
+	TestBlock block2;
+	TestTransaction tr2(trJson2);
+	block2.addTransaction(tr2);
+	blChain.addBlock(block2);
+
+	BOOST_CHECK_MESSAGE(blChain.number() == 2, "Block chain simple block import error!");
+
+	blChain.rebuild();
+
+	BOOST_CHECK_MESSAGE(blChain.number() == 2, "Block chain size after rebuild won't match!");
```

Comment:
Can you rewrite the message and use present tense? Or "Block chain size mismatch after rebuild!".